experimental/ratify/proto/v1/common.proto

```diff
@@ -0,0 +1,50 @@
+syntax="proto3";
+
+package common;
+
+option go_package = "github.com/deislabs/ratify/experimental/proto/v1/common";
+
+// ArtifactDescriptor holds various properties of an artifact.
+//
+// The only strictly defined property is that of `path`. Beyond that, it is designed to be flexible:
+// only what is known is included.
+message ArtifactDescriptor {
+    message Attributes {
+        map<string,string> values = 1;
+    }
+    // The path or address that can be used to retrieve the artifact
+    string path = 1;
+    // Collection of key/value pairs containing relevant known metadata
+    // Examples would include tag or digest.
+    repeated Attributes attributes = 2;
+}
+
+// ArtifactReference holds a string identifier and the specific metadata extracted through parsing said identifier.
+message ArtifactReference {
+    // The raw value provided to identify and/or access an artifact.
+    string rawValue = 1;
+    // Properties extracted or parsed from the raw value.
+    ArtifactDescriptor descriptor = 2;
+}
+
+// Artifact holds properties on a given artifact.
+message Artifact {
+    // The type of the artifact.
+    string artifactType = 1;
+    // All known properties (parsed or retrieved) for the artifact.
+    ArtifactDescriptor descriptor = 2;
+}
+
+// LinkedArtifact includes all relevant properties and blobs for a reference artifact.
+// LinkedArtifact is also known as a `referrer`.
+message LinkedArtifact {
+    message ArtifactBlobs {
+        repeated ArtifactDescriptor layers = 1;
```

Comment:
`layers` might a bit confusing since the "Linked Artifact" is really a manifest form of an OCI Artifact which contains blobs. It's probably not a huge deal since we don't need to conform exactly to the OCI spec for internal protos being used.